stretch_moveit_config/demo.md

```diff
@@ -0,0 +1,95 @@
+![](../images/banner.png)
+
+## Overview
+
+MoveIt 2 is a whole-body motion planning framework for mobile manipulators that allows planning pose and joint goals in environments with and without obstacles. Stretch being a mobile manipulator is uniquely well-suited to utilize the planning capabilities of MoveIt 2 in different scenarios.
+
+## Motivation
+
+Stretch has a kinematically simple 3 DoF arm (+2 with DexWrist) that is suitable for pick and place tasks of varied objects. Its mobile base provides it with 2 additional degrees of freedom that afford it more manipulability and also the ability to move around freely in its environment. To fully utilize these capabilities, we need a planner that can plan for both the arm and the mobile base at the same time. With MoveIt 2 and ROS 2, it is now possible to achieve this, empowering users to plan more complicated robot trajectories in difficult and uncertain environments.
+
+## Demo with Stretch Robot
+
+### Installing ROS 2
+By default, Stretch RE1 ships with Ubuntu 18.04, ROS Melodic, and Python2 packages. These steps will install a second operating system with Ubuntu 20.04, ROS Noetic, ROS 2 Galactic (where MoveIt 2 is supported), and Python3 packages.
+
+1. A new firmware called protocol 1 (p1) for Stretch adds support for spline trajectories, a feature used by MoveIt2. Use the [firmware update guide](https://github.com/hello-robot/stretch_firmware/blob/master/tutorials/docs/updating_firmware.md) to install the new firmware.
+
+2. Use the [Noetic installation instructions](https://github.com/hello-robot/stretch_ros/blob/dev/noetic/install_noetic.md) to create the second Ubuntu 20.04 partition, and create a ROS2 Galactic workspace with MoveIt 2 installed.
+
+3. Boot into the new Ubuntu 20.04 partition, and edit the last few lines of the ~/.bashrc file to comment out sourcing of Noetic workspaces. When both versions of ROS are sourced, package conflicts arise. The result should look like:
+
+```
+#source /opt/ros/noetic/setup.bash
+source /opt/ros/galactic/setup.bash
+#source ~/catkin_ws/devel/setup.bash
+source ~/ament_ws/install/setup.bash
+```
```

Comment:
Let's replace these steps to instead point to the [robot upgrade guide](https://docs.hello-robot.com/0.2/stretch-install/docs/robot_install/).